lego/apps/users/tests/test_student_confirmation_api.py

```diff
@@ -1,103 +1,183 @@
+from enum import Enum
 from unittest import mock
 
-from django.urls import reverse
+from django.http import HttpResponseRedirect
 from rest_framework import status
 
 from lego.apps.users import constants
 from lego.apps.users.models import AbakusGroup, User
-from lego.apps.users.registrations import Registrations
 from lego.utils.test_utils import BaseAPITestCase
 
 
-def _get_list_request_url():
-    return reverse("api:v1:student-confirmation-request-list")
+class MockFeideOAUTH:
+    _auth_url = "https://auth.mock-feide.no/auth"
 
+    def __init__(self, token="valid_token"):
+        self.token = token
 
-def _get_list_perform_url():
-    return reverse("api:v1:student-confirmation-perform-list")
+    def authorize_redirect(self, request, redirect_url):
+        return HttpResponseRedirect(self._auth_url)
 
+    def authorize_access_token(self, request):
+        return _token(self.token)
 
-def _get_student_confirmation_token_request_url(token):
-    return f"{_get_list_request_url()}?token={token}"
+    def userinfo(self, **kwargs):
+        uid = f"{kwargs.get('token')['access_token']}@ntnu.no"
+        return {"https://n.feide.no/claims/eduPersonPrincipalName": uid}
 
 
-def _get_student_confirmation_token_perform_url(token):
-    return f"{_get_list_perform_url()}?token={token}"
+mockFeide = MockFeideOAUTH()
 
 
-class RetrieveStudentConfirmationAPITestCase(BaseAPITestCase):
+class Token(Enum):
+    DATA = "data"
+    KOMTEK = "komtek"
+    DATA_MASTER = "data_midt"
+    KOMTEK_MASTER = "komtek_master"
+    MULTI_OTHER = "others"
+    INDOK = "indok"
+
+
+def _token(token):
+    return {"access_token": token}
+
+
+data_resp = [
+    {
+        "id": "fc:fs:fs:prg:ntnu.no:MTDT",
+        "type": "fc:fs:prg",
+        "displayName": "Computer Science",
+        "membership": {
+            "basic": "member",
+            "active": True,
+            "displayName": "Student",
+            "fsroles": ["STUDENT"],
+        },
+        "parent": "fc:org:ntnu.no",
+        "url": "http://www.ntnu.no/studier/mtdt",
+    }
+]
+
+komtek_resp = [
+    {
+        "id": "fc:fs:fs:prg:ntnu.no:MTKOM",
+        "type": "fc:fs:prg",
+        "displayName": "Communication Technology",
+        "membership": {
+            "basic": "member",
+            "active": True,
+            "displayName": "Student",
+            "fsroles": ["STUDENT"],
+        },
+        "parent": "fc:org:ntnu.no",
+        "url": "http://www.ntnu.no/studier/mtkom",
+    }
+]
+
+data_master_resp = [
+    {
+        "id": "fc:fs:fs:prg:ntnu.no:MIDT",
+        "type": "fc:fs:prg",
+        "displayName": "Computer Science",
+    }
+]
+
+komtek_master_resp = [
+    {
+        "id": "fc:fs:fs:prg:ntnu.no:mstcnns",
+        "type": "fc:fs:prg",
+        "displayName": "Digital Infrastructure and Cyber Security",
+    }
+]
```

Comment:
I don't see this in any of the tests, yeet or use it:)

Reply:
Oppsie, missed this. Thanks!